### PR TITLE
Fixed InkWell splash not showing up on Android

### PR DESCRIPTION
**Pull request by mrkajetanp**
Replaced Container with Ink inside SettingsList
Wrapped SettingsList in Material

Solves issue #27

Additionally deleted an auto-generated file that shouldn't be checked into git in the first place.

### CODE REVIEW

**Review by AAverin — APPROVED** [review]

File: lib/src/settings_list.dart [hunk 1]
```diff
@@ -23,30 +23,32 @@ class SettingsList extends StatelessWidget {
...
-        itemCount: sections.length,
-        itemBuilder: (context, index) {
-          AbstractSection current = sections[index];
-          AbstractSection futureOne;
-          try {
-            futureOne = sections[index + 1];
-          } catch (e) {}
+    return Material(
+      child: Ink(
+        color: Theme.of(context).brightness == Brightness.light
```
Comment (AAverin) [inline lib/src/settings_list.dart:28]: color should be probably set on Material, not on Ink

**Comment by AAverin** [comment]
Ok, I just noticed it changes not the Tile, but List.
Shouldn't the Ink be applied to the SettingsTile? Why have it on the SettingsList?

**Comment by AAverin** [comment]
This did it for me so far:
```
class InkSettingsTile extends SettingsTile {
  final String title;
  final String subtitle;
  final Widget leading;
  final Widget trailing;
  final VoidCallback onTap;
  final bool enabled;
  final TextStyle titleTextStyle;
  final TextStyle subtitleTextStyle;
  final Color switchActiveColor;

  const InkSettingsTile({
    Key key,
    @required this.title,
    this.subtitle,
    this.leading,
    this.trailing,
    this.onTap,
    this.titleTextStyle,
    this.subtitleTextStyle,
    this.enabled = true,
    this.switchActiveColor,
  }) : super(key: key);

  @override
  Widget build(BuildContext context) {
    return Material(
      child: Ink(
        color: Theme.of(context).brightness == Brightness.light ? Color(0xFFEFEFF4) : Colors.black,
        child: super.build(context),
      ),
    );
  }
}
```

**Comment by kdrag0n** [comment]
A cleaner and more complete version of the wrapped SettingsTile above, as an interim solution until this PR gets merged:

```dart
class MaterialSettingsTile extends SettingsTile {
  const MaterialSettingsTile({
    Key key,
    @required String title,
    String subtitle,
    Widget leading,
    Widget trailing,
    VoidCallback onTap,
    TextStyle titleTextStyle,
    TextStyle subtitleTextStyle,
    bool enabled = true,
    Color switchActiveColor,
  }) : super(
    key: key,
    title: title,
    subtitle: subtitle,
    leading: leading,
    trailing: trailing,
    onTap: onTap,
    titleTextStyle: titleTextStyle,
    subtitleTextStyle: subtitleTextStyle,
    enabled: enabled,
    switchActiveColor: switchActiveColor,
  );

  const MaterialSettingsTile.switchTile({
    Key key,
    @required String title,
    String subtitle,
    Widget leading,
    Widget trailing,
    @required Function(bool value) onToggle,
    @required bool switchValue,
    TextStyle titleTextStyle,
    TextStyle subtitleTextStyle,
    bool enabled = true,
    Color switchActiveColor,
  }) : super.switchTile(
    key: key,
    title: title,
    subtitle: subtitle,
    leading: leading,
    trailing: trailing,
    onToggle: onToggle,
    switchValue: switchValue,
    titleTextStyle: titleTextStyle,
    subtitleTextStyle: subtitleTextStyle,
    enabled: enabled,
    switchActiveColor: switchActiveColor,
  );

  @override
  Widget build(BuildContext context) {
    return Material(
      child: Ink(
        child: super.build(context),
      )
    );
  }
}
```

I think wrapping the entire SettingsList in Material + Ink will cause ink splashes to render outside the bounds of the tile that is being interacted with, as well as causing the entire screen to get highlighted if you hold long enough.

File: lib/src/settings_list.dart [hunk 2]
```diff
@@ -23,30 +23,32 @@ class SettingsList extends StatelessWidget {
...
-        shrinkWrap: shrinkWrap,
-        itemCount: sections.length,
-        itemBuilder: (context, index) {
-          AbstractSection current = sections[index];
-          AbstractSection futureOne;
-          try {
-            futureOne = sections[index + 1];
-          } catch (e) {}
+    return Material(
+      child: Ink(
```
Comment (Akora-IngDKB) [inline lib/src/settings_list.dart:27]: Wrapping the `ListView` in an `Ink` will cause the splash to be applied all across the list as opposed to just a single tile.

**Review by Akora-IngDKB — CHANGES_REQUESTED** [review]
I think you should consider doing this on the `SettingsTile` instead of the `SettingsList`.

**Comment by Akora-IngDKB** [comment]
> A cleaner and more complete version of the wrapped SettingsTile above, as an interim solution until this PR gets merged:
> 
> ```dart
> class MaterialSettingsTile extends SettingsTile {
> ......
>   @override
>   Widget build(BuildContext context) {
>     return Material(
>       child: Ink(
>         child: super.build(context),
>       )
>     );
>   }
> }
> ```
> 
> I think wrapping the entire SettingsList in Material + Ink will cause ink splashes to render outside the bounds of the tile that is being interacted with, as well as causing the entire screen to get highlighted if you hold long enough.

This looks like good.

**Pull request merged**